### PR TITLE
AIRFLOW-3437 Adding better json

### PR DESCRIPTION
**Pull request by PythonGirlSam**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-3437) issues and references them in the PR title.
  - https://issues.apache.org/jira/browse/AIRFLOW-3437

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Current rest API "/api/experimental/dags/<dag_id>dag_runs"returns message like below, which makes difficult for developer to figure out execution date/run_id and execution date/run_id extract logic has to be written  

{ "message": "Created <DagRun example_bash_operator @ 2018-12-03 11:16:36+00:00: manual__2018-12-03T11:16:36+00:00, externally triggered: True>" 
### Tests

- [x] My PR adds the following unit tests :

- test_trigger_dag_detailed_output 

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by PythonGirlSam** [comment]
Under PR https://github.com/apache/incubator-airflow/pull/4271 I got below comment
"Additionally you need tests - I think you end up with "start_date": True in your output". I have well tested this start_date wont be True but rather empty in case it is not fetched or if formate is not correct.

File: airflow/www/api/experimental/endpoints.py [hunk 1]
```diff
@@ -132,8 +142,8 @@ def task_info(dag_id, task_id):
 
 
 @api_experimental.route(
-    '/dags/<string:dag_id>/dag_runs/<string:execution_date>/tasks/<string:task_id>',
-    methods=['GET'])
+    '/dags/<string:dag_id>/dag_runs/<string:execution_date>/' +
+    'tasks/<string:task_id>', methods=['GET'])
```
Comment (phoenix24) [inline airflow/www/api/experimental/endpoints.py:146]: doesn't look like a relevant change.

File: airflow/www/api/experimental/endpoints.py [hunk 2]
```diff
@@ -189,7 +199,8 @@ def latest_dag_runs():
                 'dag_run_url': url_for('airflow.graph', dag_id=dagrun.dag_id,
                                        execution_date=dagrun.execution_date)
             })
-    return jsonify(items=payload)  # old flask versions dont support jsonifying arrays
+    # old flask versions dont support jsonifying arrays
+    return jsonify(items=payload)
```
Comment (phoenix24) [inline airflow/www/api/experimental/endpoints.py:203]: doesn't look like a relevant change.

**Review by phoenix24 — APPROVED** [review]

**Comment by dwai1714** [comment]
@holdenk See this ?

File: airflow/www/api/experimental/endpoints.py (hunk 1 above)
Comment (ashb) [reply airflow/www/api/experimental/endpoints.py:146]: Agreed. Revert this please

File: airflow/api/common/experimental/get_dag_runs.py [hunk 3]
```diff
@@ -50,6 +54,4 @@ def get_dag_runs(dag_id, state=None):
                            run.start_date.isoformat()),
             'dag_run_url': url_for('Airflow.graph', dag_id=run.dag_id,
                                    execution_date=run.execution_date)
-        })
-
-    return dag_runs
```
Comment (ashb) [inline airflow/api/common/experimental/get_dag_runs.py:55]: The change in this file serves no purpose and just confuses git history. Please revert this.

File: airflow/www/api/experimental/endpoints.py [hunk 4]
```diff
@@ -66,7 +71,8 @@ def trigger_dag(dag_id):
         except ValueError:
             error_message = (
                 'Given execution date, {}, could not be identified '
-                'as a date. Example date format: 2015-11-16T14:34:15+00:00'.format(
+                'as a date. Example date format: '
+                '2015-11-16T14:34:15+00:00'.format(
```
Comment (ashb) [inline airflow/www/api/experimental/endpoints.py:75]: Revert this change please.

File: airflow/www/api/experimental/endpoints.py [hunk 5]
```diff
@@ -84,8 +90,11 @@ def trigger_dag(dag_id):
 
     if getattr(g, 'user', None):
         _log.info("User {} created {}".format(g.user, dr))
-
-    response = jsonify(message="Created {}".format(dr))
+    if detailed_output:
+        dr = format_dag_run(dr)
+        response = jsonify(**dr)
+    else:
+        response = jsonify(message="Created {}".format(dr))
```
Comment (ashb) [inline airflow/www/api/experimental/endpoints.py:97]: Why  not have both?

```
return  jsonify(message="Created {}".format(dr), **dr)
```

That makes it easier to deprecate  the old message over time without also having to add special handling of the new parameter.

**Review by ashb — CHANGES_REQUESTED** [review]
Lots of formatting only changes that should be un-done, and I think we can remove the extra query parameter.
